src/ucp/tag/eager_rcv.c

```diff
@@ -279,10 +279,15 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_eager_sync_ack_handler,
 {
     ucp_worker_h    worker   = arg;
     ucp_reply_hdr_t *rep_hdr = data;
-    ucp_request_t   *req     = ucp_worker_extract_request_by_id(worker,
-                                                                rep_hdr->req_id);
+    ucp_request_t *req;
 
-    ucp_tag_eager_sync_completion(req, UCP_REQUEST_FLAG_REMOTE_COMPLETED, UCS_OK);
+    if (worker->context->config.ext.proto_enable) {
+        ucp_proto_eager_sync_ack_handler(worker, rep_hdr);
+    } else {
+        req = ucp_worker_extract_request_by_id(worker, rep_hdr->req_id);
+        ucp_tag_eager_sync_completion(req, UCP_REQUEST_FLAG_REMOTE_COMPLETED,
+                                      UCS_OK);
```

Comment:
can this code be united? Seems  no need to separate to different flows

Reply:
i'd like to remove 2nd flow eventually, and it would be easier if they are separated

Reply:
ok